board/drivers/can.h

```diff
@@ -65,7 +72,7 @@ int can_err_cnt = 0;
 int can_overflow_cnt = 0;
 
 // ********************* interrupt safe queue *********************
-
+// cppcheck-suppress misra-c2012-8.2
```

Comment:
to fix all these, we could declare the prototypes somewhere on top in this file

Reply:
Prototypes are already declared. It stops complaining if you will remove prototypes, but then you need to shuffle some funcs. Adeeb told to do that as a second PR as it looks massive.